omnigibson/scenes/scene_base.py

```diff
@@ -169,85 +205,138 @@ def object_registry_group_keys(self):
     def loaded(self):
         return self._loaded
 
+    @property
+    def idx(self):
+        """Index of this scene in the simulator. Should not change."""
+        # TODO(parallel-hang): Return self._idx
+        return og.sim.scenes.index(self)
+
     @property
     def initialized(self):
         return self._initialized
 
+    @property
+    def use_floor_plane(self):
+        return self._use_floor_plane
+
+    def prebuild(self):
+        """
+        Prebuild the scene USD before loading it into the simulator. This is useful for caching the scene USD for faster
+        loading times.
+
+        Returns:
+            str: Path to the prebuilt USD file
+        """
+        # Prebuild and cache the scene USD using the objects
+        if self.scene_file not in PREBUILT_USDS:
+            # Prebuild the scene USD
+            log.info(f"Prebuilding scene file {self.scene_file}...")
+
+            # Create a new stage inside the tempdir, named after this scene's file.
+            decrypted_fd, usd_path = tempfile.mkstemp(os.path.basename(self.scene_file) + ".usd", dir=og.tempdir)
```

Comment:
I'm guessing we have to encrypt our USDs because they include the encrypted object assets?

Also, is this actually portable? e.g.: if I save a scene file and then send it to someone else, can it be loaded on their machine properly? Previously we were unable to do this